File: src/planetaris.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){
    int n, a; cin >> n >> a;
    int ships[n];
    for(int i = 0; i < n; i++){
        cin >> ships[i];
    }

    sort(ships, ships + n);

    int w = 0;
    for(int i = 0; i < n; i++){
        if(a > ships[i]){
            w++; a = a - (ships[i] + 1);
        }
        else break;
    }
    cout << w << endl;

    return 0;
}

File: src/planetaris.py
def maxwon(n, a, ships):
    ships.sort()
    w = 0
    for i in range(n):
        if a > ships[i]:
            w += 1
            a -= (ships[i] + 1)
        else: break
    return w

n, a = map(int, input().split())
ships = list(map(int, input().split()))

print(maxwon(n, a, ships))